# _WAS

Any [_] Wide Association Studies - software issues and solutions.

## System setup

1. It is relevant to have knowledge about GNU parallel and sge or slurm (e.g., https://github.com/statgen/SLURM-examples); see also descriptions in other pipelines here.
2. Here is a section dedicated to [R and RStudio](R).
3. A broad coverage on genomewide association studies is available from [vdi.md](https://github.com/jinghuazhao/GDCT/blob/master/vdi.md) as part of the removed GWAS-2017 repository and listed in [GDCT repository](https://github.com/jinghuazhao/GDCT).

## An extensible list of applications

The repository also contains extensive information complementary but not exclusive to a number of pipelines on the GitHub.

### Single variant analysis

* [PLINK2](PLINK2)
* [PyLMM](PyLMM)

common, exome, rare, models, plots

### Finemapping

* [JAM](JAM)

### Pathway analysis

* [DEPICT](DEPICT)
* [PASCAL](PASCAL)

### Medelian randomiszation

* [R-packages](R-packages)

### Polygenic prediction

* [R-packages](R-packages)

### PheWAS

* [PheWAS information](PheWAS)
* [R-packages](R-packages)

### Transcriptome-wide association analysis (TWAS)

* [MetaXcan](MetaXcan) / S-PrediXcan

eQTL, single cell sequencing, epigenome-wide association study (EWAS), therapeutic targets
