# software-notes

**Software notes for [_]Wide Association Studies**

This repository collects information on **[overall setup](overall)** and specific aspects for software. It evolves over time; additional information is also kept in 
[NOTES](NOTES.md). It is envisaged that the site could eventually be enriched by many practitioners.

### Single variant analysis

* [PLINK2](PLINK2)
* [PyLMM](PyLMM)

### HLA imputation

* [IMP*HLA](IMPHLA02)

### Exome and rare variants

* [Jannovar](jannovar)

### Finemapping

* [CAVIAR/eCAVIAR](caviar)
* [JAM](JAM)

### Functional annotation

* [R-packages](R-packages)

### Pathway analysis

* [DEPICT](DEPICT) see the latest [GIANT+Biobank BMI analysis](https://github.com/jinghuazhao/Omics-analysis/tree/master/BMI) ![#f03c15](https://placehold.it/15/f03c15/000000?text=+) 
* [PASCAL](PASCAL)

### Medelian randomiszation

* [R-packages](R-packages)

### NGS

* [GATK](GATK)
* [IGV](IGV)
* [VarScan](VarScan)

### Polygenic modeling

* [ldsc](ldsc)
* [R-packages](R-packages)

### PheWAS

* [Information](PheWAS)
* [R-packages](R-packages)

### Transcriptome-wide association analysis (TWAS)

* [MetaXcan](MetaXcan) / S-PrediXcan
* [R-packages](R-packages)

eQTL, single cell sequencing, epigenome-wide association study (EWAS), therapeutic targets
